cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexManagerImpl.java

```diff
@@ -256,6 +257,32 @@ public String ensureIndexed(List<FieldSort> fieldNames,
                                 String indexName,
                                 IndexType indexType,
                                 String tokenize) throws QueryException {
+
+        // If the index name is null, a name should be generated, but first we should check
+        // if an index already exists that matches the request index definition.
+        if (indexName == null) {
+            List<Index> indexes = this.listIndexes();
+            Collections.sort(fieldNames);
```

Comment:
I think we need to create a mutable copies of the lists here before we sort them, as this crashes if the list is immutable.  The tests seem to be currently failing for this reason because they use `Collections.singletonList()`, which creates an immutable list [here](https://github.com/cloudant/sync-android/blob/cb057b68c12a517370ce9df5a5fd3fe6d01198be/cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/IndexManagerTest.java#L46)